prusti-specs/src/specifications/preparser.rs

```diff
@@ -161,6 +161,17 @@ impl PrustiTokenStream {
         self.tokens.is_empty()
     }
 
+    fn parse_rest<T, F>(mut self, f: F) -> syn::Result<T>
+    where
+        F: FnOnce(&mut Self) -> syn::Result<T>,
+    {
+        let result = f(&mut self)?;
+        if !self.is_empty() {
+            return error(self.source_span, "unexpected extra tokens");
```

Comment:
You can use the span of the first token, since you know there is at least one left.